python/sources/simple-csv-new/connector/source_connector.py

```diff
@@ -0,0 +1,178 @@
+import csv
+import logging
+import json
+from typing import Optional, Union, Callable, Any, Iterator
+
+from .connector_templates import (
+    SourceConsumer,
+    SourceProducer,
+    SourceConnector,
+    setup_logging
+)
+from quixstreams.logging import LogLevel
+from quixstreams.models.messages import KafkaMessage
+from quixstreams.models.serializers import SerializerType
+from quixstreams.models.types import MessageKey, MessageValue
+
+
+logger = logging.getLogger(__name__)
+
+__all__ = ("CsvSourceConnector",)
+
+
+class CsvConsumer(SourceConsumer):
+    """
+    A thin wrapper around csv.DictReader for lazy row-based reading of a CSV file
+    with a header.
+    """
+
+    def __init__(
+        self,
+        path: str,
+        key_parser: Optional[Union[str, Callable[[dict], MessageKey]]] = None,
```

Comment:
I think the word `extractor` would better describe what it does.
To me `parser` is associated with parsing data from some string representation, e.g. "XML parser", "HTTP parser", etc

Reply:
I can see the thought, will change